### DISCUSSION TITLE
Can I reduce the execution time of functions in nextJS??

### DISCUSSION THREAD

**Discussion by judy-D** [vercel/next.js#22630] [Help]
Hello, 

I have a website with NextJS front-end, with Strapi backend, I deployed this website,
but sometimes while navigating between pages, I get this error:

{“errorMessage”:“2021-02-28T07:01:23.806Z 4234f202-e15d-4a25-84ce-79b9c82ac634 Task timed out after 10.01 seconds”}

I have 6 functions, I heard someone said that if I reduced the execution time of the function that could the error?

How can I reduce it? is there any way to this error? what is the best way to deploy NextJs if I have a number of functions?? 

Thanks

**Comment by michrome** [comment]
Assuming you're deploying to Vercel, on the free plan the [Serverless Function Execution Timeout is **10 seconds**]( https://vercel.com/docs/platform/limits?query=limits#serverless-function-execution-timeout). The [Pro plan](https://vercel.com/pricing) will give you up to 60 seconds.

You'll need to profile your function's code to understand what's running slowly, so you can optimise it or perhaps use a different architectural approach: your users won't thank you if you're making them wait multiple seconds while navigating between pages!
